Add dimension mapping parameter (#74450)
elastic/elasticsearch: docs/reference/mapping/types/numeric.asciidoc

    Accepts `true` (default) and `false`. Not applicable for `unsigned_long`.
    Note that this cannot be set if the `script` parameter is used.

`dimension`::

>     Marks the field as a time series dimension. Accepts `true` or `false`
>     (default). Only `byte`, `short`, `integer`, and `long` numeric fields
>     support this parameter.

Reviewer: ```suggestion
    For internal use only. Marks the field as a time series dimension in {kib}.
    Accepts `true` or `false` (default). Only `byte`, `short`, `integer`, and
    `long` numeric fields support this parameter.
```

> <<dimension,`dimension`>>::
> 
>     This parameter marks the field as a time series dimension field. Accepts
>     `true` or `false` (default). This parameter can be set only for fields
>     of type `byte`, `short`, `integer` and `long`.

Reviewer: ```suggestion
`dimension`::

    Marks the field as a time series dimension. Accepts `true` or `false`
    (default). Only `byte`, `short`, `integer`, and `long` numeric fields
    support this parameter.
```
